accepted/2022/version-selection.md

```diff
@@ -0,0 +1,449 @@
+
+This feature can be thought of as an analog to `global.json` SDK version selection, but for controlling the runtime version. Thinking of it that ways helps contextualize that this feature is another case of creating parity between SDK and runtime selection features.
+
+An organization could alternatively use `DOTNET_ROLL_FORWARD=Major`. On the upside, this capability already exists and would provide much of the same behavior. On the downside, roll-forward affects all versions, and could mask some deployment problems that should be resolved. For example, the organization might intend to install both .NET Core 3.1 and .NET 6, but accidentally only install .NET 6. The roll-forward configuration could mask that issue. Even if it worked without issue, it isn't something you want without it being intentional.
+
+Note: The `Major` roll-forward policy is only applied when the requested major version is NOT available. It doesn't force applications to use a later version when the preferred version is available.
+
+Note: Assuming we enable this scenario, the .NET application framework teams (like ASP.NET Core) would need to focus more on binary upgrade compatibility.
+
+### Simplifying runtime roll-forward gestures
+
+The following changes don't directly contribute to the customer scenarios described earlier, but are intended to improve overall developer experience.
+
+The `--roll-forward` runtime switch currently requires a value. We have an opportunity to make this functionality simpler to use by offering a generally useful policy default when you specify the switch with no value. The obvious option is `Major`. This change would likely encourage the use of the roll-forward capability. The SDK doesn't need such an experience since it already implements the `LatestMajor` policy by default.
+
+That would enable the following gesture:
+
+```bash
+app --roll-forward
+```
+
+It would be identical to the following examples:
+
+```bash
+app --roll-forward Major
+app --roll-forward Major -- --myarg 1234
+```
+
+Note: If your app also accepts its own arguments, you can use `--` (it is an industry convention) to separate host from app arguments.
+
+### Support ranges
+
+Some folks have asked us for a way to state compatibility for a range of versions, for an app. It's a good scenario, but not a feature we've designed or invested in.
+
+## Implemenation Plan
+
+There are several (some unrelated) changes specified in this document. The following is proposed as the order (and grouping) to approach them.
+
+Priority 1:
+
+- Disable MLL.
+- Add SDK ENVs.
+- Change default for `global.json` SDK version to a floor version.
+
+Priority 2:
+
+- Make SDK and Runtime gestures consistent, complete, and correct (fix `--fx-version`, add `--sdk-version`, ...).
```

Comment:
We need to make sure all cli args and switches can be codified, so Windows users can invoke apps by double clicking on exe and not use cli.

Reply:
Sure. For the most part, they already are. Most of the spec is about expending SDK file-based config to other places. The only one (that I'm aware of) to make VS work is the `.dotnet` directory.